startup error: `panic: interface conversion: interface {} is int, not string`

## Bug Report

### Description

With `v4.34.X`, the container does not start but a goroutine panics. It is possible that my configuration is not suited for the new minor version, but with `v4.33.X` all is well.

### Expected Behaviour

Authelia should start.

### Reproduction Steps

``` CONSOLE
$ docker run --rm -v $(pwd)/configuration.yml:/config/configuration.yml authelia/authelia:4.34.2
panic: interface conversion: interface {} is int, not string

goroutine 1 [running]:
github.com/authelia/authelia/v4/internal/configuration.ToTimeDurationFunc.func1({0x55ee37ed4920, 0x55ee37d03bc0}, {0x55ee37ed4920, 0x55ee37dfcb80}, {0x55ee37d03bc0, 0x55ee385d3f60})
	github.com/authelia/authelia/v4/internal/configuration/decode_hooks.go:88 +0x3a9
github.com/mitchellh/mapstructure.DecodeHookExec({0x55ee37d737a0, 0x55ee37e87438}, {0x55ee37d03bc0, 0x55ee385d3f60, 0x55ee37d48c20}, {0x55ee37dfcb80, 0xc0001356f8, 0x55ee37d6b6c0})
	github.com/mitchellh/mapstructure@v1.4.3/decode_hooks.go:47 +0x190
github.com/mitchellh/mapstructure.ComposeDecodeHookFunc.func1({0x55ee37d03bc0, 0x55ee385d3f60, 0xc0005be368}, {0x55ee37dfcb80, 0xc0001356f8, 0x0})
	github.com/mitchellh/mapstructure@v1.4.3/decode_hooks.go:69 +0x10c
github.com/mitchellh/mapstructure.DecodeHookExec({0x55ee37d48c20, 0xc0004a2d20}, {0x55ee37d03bc0, 0x55ee385d3f60, 0x14}, {0x55ee37dfcb80, 0xc0001356f8, 0xc000314080})
	github.com/mitchellh/mapstructure@v1.4.3/decode_hooks.go:51 +0xd3
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000010640, {0xc000314080, 0x55ee36908005}, {0x55ee37d03bc0, 0x55ee385d3f60}, {0x55ee37dfcb80, 0xc0001356f8, 0xd})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:440 +0x179
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000010640, {0x55ee37caa256, 0xc0001356a8}, {0x55ee37d52a60, 0xc00049b5c0, 0x2}, {0x55ee37e28be0, 0xc0001356a8, 0xc00049b290})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:1382 +0x14b0
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0x55ee37d48c20, {0x55ee37caa256, 0x7}, {0x55ee37d52a60, 0xc00049b5c0}, {0x55ee37e28be0, 0xc0001356a8, 0x55ee35dd7714})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:1208 +0x4b5
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000010640, {0x55ee37caa256, 0x55ee36908005}, {0x55ee37d52a60, 0xc00049b5c0}, {0x55ee37e28be0, 0xc0001356a8, 0x10})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:463 +0x3d9
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000010640, {0x0, 0xc000135600}, {0x55ee37d52a60, 0xc00049b2c0, 0x2}, {0x55ee37e4fca0, 0xc000135600, 0xc00049b290})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:1382 +0x14b0
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0x55ee37d48c20, {0x0, 0x0}, {0x55ee37d52a60, 0xc00049b2c0}, {0x55ee37e4fca0, 0xc000135600, 0xc0004b4000})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:1208 +0x4b5
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000010640, {0x0, 0xc000490e01}, {0x55ee37d52a60, 0xc00049b2c0}, {0x55ee37e4fca0, 0xc000135600, 0xc000492db0})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:463 +0x3d9
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc000010640, {0x55ee37d52a60, 0xc00049b2c0})
	github.com/mitchellh/mapstructure@v1.4.3/mapstructure.go:398 +0xd8
github.com/knadh/koanf.(*Koanf).UnmarshalWithConf(0xc00029fb00, {0x0, 0x0}, {0x55ee37cd3fe0, 0xc000135600}, {{0x0, 0x55ee37eab310}, 0x0, 0xc000482320})
	github.com/knadh/koanf@v1.4.0/koanf.go:269 +0x313
github.com/authelia/authelia/v4/internal/configuration.unmarshal(0x2c0, 0xc00029f7a0, {0x0, 0x0}, {0x55ee37cd3fe0, 0xc000135600})
	github.com/authelia/authelia/v4/internal/configuration/provider.go:56 +0x1e6
github.com/authelia/authelia/v4/internal/configuration.LoadAdvanced(0xc00029f7a0, {0x0, 0x0}, {0x55ee37cd3fe0, 0xc000135600}, {0xc0002a3300, 0x3, 0x4})
	github.com/authelia/authelia/v4/internal/configuration/provider.go:37 +0x18d
github.com/authelia/authelia/v4/internal/configuration.Load(...)
	github.com/authelia/authelia/v4/internal/configuration/provider.go:16
github.com/authelia/authelia/v4/internal/commands.loadConfig({0xc000291580, 0x1, 0x1}, 0x1, 0x1)
	github.com/authelia/authelia/v4/internal/commands/configuration.go:84 +0x131
github.com/authelia/authelia/v4/internal/commands.newCmdWithConfigPreRun.func1(0xc0002c3680, {0xc00027f120, 0x2, 0x2})
	github.com/authelia/authelia/v4/internal/commands/configuration.go:56 +0x1d6
github.com/spf13/cobra.(*Command).execute(0xc0002c3680, {0xc0000320a0, 0x2, 0x2})
	github.com/spf13/cobra@v1.3.0/command.go:849 +0x583
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002c3680)
	github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.3.0/command.go:902
main.main()
	github.com/authelia/authelia/v4/cmd/authelia/main.go:10 +0x1e
```

with the following `configuration.yml` file:

``` YAML
# sbev! == set by environment variables
---
access_control:
  default_policy: deny
  rules:
    - domain:
      - x.x.com
      policy: bypass

    - domain:
        - x.x.com
      policy: one_factor
      subject:
        - user:x.x
        - user:y.y
        - user:z.z

    - domain:
        - x.com
      policy: one_factor
      subject:
        - user:x.x
        - user:y.y

    - domain: '*.y.com'
      policy: two_factor
      subject: user:x.x

authentication_backend:
  disable_reset_password: true
  refresh_interval: 5m
  file:
    path: /config/users-database.yml
    password:
      algorithm: argon2id
      iterations: 1
      parallelism: 6
      salt_length: 16
      memory: 128

duo_api:
  hostname: api-x.duosecurity.com
  integration_key: XXXXXXX
  enable_self_enrollment: false
  # secret_key: sbev! AUTHELIA_DUO_API_SECRET_KEY_FILE

identity_providers:
  oidc:
    access_token_lifespan: 1h
    authorize_code_lifespan: 1m
    id_token_lifespan: 1h
    refresh_token_lifespan: 90m

    enable_client_debug_messages: false
    # enforce_pkce: public_clients_only
    # enable_pkce_plain_challenge: false

    # hmac_secret: sbev! AUTHELIA_IDENTITY_PROVIDERS_OIDC_HMAC_SECRET_FILE
    # issuer_private_key: sbev! AUTHELIA_IDENTITY_PROVIDERS_OIDC_ISSUER_PRIVATE_KEY_FILE

    clients:
      - id: gitlab
        description: GitLab
        public: false
        secret: xxxx
        authorization_policy: one_factor
        audience:
          - user:x.x
          - user:y.y
        redirect_uris:
          - https://x.x.com/users/auth/openid_connect/callback

      # - id: owncloud
      #   description: ownCloud

log:
  level: warn
  format: text

# default_redirection_url:
# jwt_secret: sbev! AUTHELIA_JWT_SECRET_FILE

notifier:
  disable_startup_check: true
  smtp:
    host: 1.1.1.1
    port: 465
    timeout: 10s
    subject: '[Authelia] {title}'
    sender: x@y.com
    username: x@y.com
    # password: sbev! AUTHELIA_NOTIFIER_SMTP_PASSWORD_FILE
    identifier: x.y.com
    startup_check_address: x@y.com
    disable_require_tls: false
    disable_html_emails: false
    tls:
      server_name: x.y.com
      skip_verify: false
      minimum_version: TLS1.3

ntp:
  address: time.cloudflare.com:123
  version: 3
  max_desync: 2s
  disable_startup_check: false
  disable_failure: false

regulation:
  ban_time: 10d
  find_time: 1h
  max_retries: 3

server:
  host: 0.0.0.0
  port: 8080
  disable_healthcheck: true

session:
  name: authelia_session
  domain: xcom
  same_site: strict
  inactivity: 30m
  expiration: 1h
  remember_me_duration: 0
  # secret: sbev! AUTHELIA_SESSION_SECRET_FILE

storage:
  # encryption_key: sbev! AUTHELIA_STORAGE_ENCRYPTION_KEY_FILE
  local:
    path: /config/db.sqlite3

theme: auto

# webauthn:
#   disable: true

# totp:
#   disable: true
```

### Additional Information

Error originates here: https://github.com/authelia/authelia/blob/8f05846e214df843ad8b996525b65ebef02a5686/internal/configuration/decode_hooks.go#L39-L101

on line 88. The conversion panics.

--- a/internal/configuration/decode_hooks.go
+++ b/internal/configuration/decode_hooks.go
@@ -67,7 +67,11 @@ func ToTimeDurationFunc() mapstructure.DecodeHookFuncType {
 
 		switch {
 		case f.Kind() == reflect.String:
-			break
+			dataStr := data.(string)
+
+			if duration, err = utils.ParseDurationString(dataStr); err != nil {
+				return nil, err
+			}
 		case f.Kind() == reflect.Int:
 			seconds := data.(int)
 
@@ -82,14 +86,6 @@ func ToTimeDurationFunc() mapstructure.DecodeHookFuncType {
 			seconds := data.(int64)
 
 			duration = time.Second * time.Duration(seconds)
-		}
-
-		if duration == 0 {
-			dataStr := data.(string)
-
-			if duration, err = utils.ParseDurationString(dataStr); err != nil {
-				return nil, err
-			}
 		}
 
 		if ptr {

--- a/internal/configuration/decode_hooks_test.go
+++ b/internal/configuration/decode_hooks_test.go
@@ -268,3 +268,25 @@ func TestToTimeDurationFunc_ShouldNotParse_FromBool(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, from, result)
 }
+
+func TestToTimeDurationFunc_ShouldParse_FromZero(t *testing.T) {
+	hook := ToTimeDurationFunc()
+
+	var (
+		from     = 0
+		expected = time.Duration(0)
+
+		to     time.Duration
+		ptrTo  *time.Duration
+		result interface{}
+		err    error
+	)
+
+	result, err = hook(reflect.TypeOf(from), reflect.TypeOf(to), from)
+	assert.NoError(t, err)
+	assert.Equal(t, expected, result)
+
+	result, err = hook(reflect.TypeOf(from), reflect.TypeOf(ptrTo), from)
+	assert.NoError(t, err)
+	assert.Equal(t, &expected, result)
+}

